### PR TITLE
#170947576 user is able to like/unlike

### PR DESCRIPTION
**Pull request by Ntare22**
#### What does this PR do?
A user is able to like or unlike a facility
#### Description of Task to be completed?
- added the likes column
- added middleware to verify facility
- added to like and unlike methods in the facilitiesController
#### How should this be manually tested?
- clone this repo
- run `npm install` to install all dependencies
- run `npm run dev`
- login in with a seeded user `jdev@andela.com`, password `Bien@BAR789`
- in the headers include that token.
- In the query params, include an id of a seeded facility `5be72db7-5510-4a50-9f15-e23f103116d5` to look like this `localhost:8000/api/v1/facilities/like?id=5be72db7-5510-4a50-9f15-e23f103116d5`
- click send while using the patch request
#### Any background context you want to provide?
- N/A
#### What are the relevant pivotal tracker stories?
[#170947576](https://www.pivotaltracker.com/story/show/170947576)
#### Screenshots (if appropriate)
image
image
image

### CODE REVIEW

**Comment by bbaime98** [comment]
can you please add screenshots?

**Review by Baraka-Mugisha — DISMISSED** [review]

**Comment by Ntare22** [comment]
@bbaime98 I've added some screenshots.

**Review by bbaime98 — DISMISSED** [review]

**Review by izzett222 — APPROVED** [review]

**Review by ceelogre — CHANGES_REQUESTED** [review]
Refactor `facilitiesController` to reduce complexity and add unit tests.

File: src/services/localesServices/locales/en.json [hunk 1]
```diff
@@ -57,5 +57,20 @@
...
+	"user should have manager before performing this operation": "user should have manager before performing this operation",
+	"The user is already a %s": "The user is already a %s",
+	"relation \"Users\" does not exist": "relation \"Users\" does not exist",
+	"No token provided": "No token provided",
+	"facility with this id does not exist": "facility with this id does not exist",
+	"user has liked facility": "user has liked facility",
+	"user has already liked facility": "user has already liked facility",
+	"user has unliked facility": "user has unliked facility",
+	"user has already unliked facility": "user has already unliked facility"
 }
```
Comment (izzett222) [inline src/services/localesServices/locales/en.json]: could you add an empty line for this file @Ntare22

File: src/controllers/facilitiesController.js [hunk 2]
```diff
@@ -82,5 +83,96 @@ class FacilitiesController {
...
+      }
+
+      if (alreadyLiked === undefined) {
+        await db.Facilities.increment('likes', { where: { id } });
+        await db.Facilities.update({
+          likesId: Sequelize.fn('array_append', Sequelize.col('likesId'), user.id)
+        }, { where: { id } });
+        return Response.success(res, 200, res.__('user has liked facility'));
+      }
+      return Response.errorResponse(res, 403, res.__('user has already liked facility'));
```
Comment (izzett222) [inline src/controllers/facilitiesController.js]: hey @Ntare22, shouldn't the like be removed when the user tries to like it two times

File: src/controllers/facilitiesController.js [hunk 3]
```diff
@@ -82,5 +83,96 @@ class FacilitiesController {
...
+      }
+
+      if (alreadyUnliked === undefined) {
+        await db.Facilities.increment('unlikes', { where: { id } });
+        await db.Facilities.update({
+          unlikesId: Sequelize.fn('array_append', Sequelize.col('unlikesId'), user.id)
+        }, { where: { id } });
+        return Response.success(res, 200, res.__('user has unliked facility'));
+      }
+      return Response.errorResponse(res, 403, res.__('user has already unliked facility'));
```
Comment (izzett222) [inline src/controllers/facilitiesController.js]: hey @Ntare22, shouldn't the unlike be removed when the user tries to unlike it two times

**Review by izzett222 — CHANGES_REQUESTED** [review]

**Review by izzett222 — DISMISSED** [review]

**Review by bbaime98 — DISMISSED** [review]

**Review by Baraka-Mugisha — DISMISSED** [review]

**Review by ceelogre — APPROVED** [review]

**Review by izzett222 — APPROVED** [review]

**Review by bbaime98 — APPROVED** [review]

**Review by Baraka-Mugisha — APPROVED** [review]

**Pull request merged**